src/sagemaker/estimator.py

```diff
@@ -282,8 +282,14 @@ def deploy(self, initial_instance_count, instance_type, endpoint_name=None, **kw
     @property
     def model_data(self):
         """str: The model location in S3. Only set if Estimator has been ``fit()``."""
-        return self.sagemaker_session.sagemaker_client.describe_training_job(
-            TrainingJobName=self.latest_training_job.name)['ModelArtifacts']['S3ModelArtifacts']
+        try:
+            model_uri = self.sagemaker_session.sagemaker_client.describe_training_job(
+                TrainingJobName=self.latest_training_job.name)['ModelArtifacts']['S3ModelArtifacts']
+        except AttributeError:
+            logging.warning('No finished training job found associated with this estimator. Please make sure'
+                            'this estimator is only used for building workflow config')
+            model_uri = os.path.join(self.output_path, self._current_job_name, 'output/model.tar.gz')
```

Comment:
if you're already using `os.path.join`, don't also hardcode a separator (i.e. '/')

Reply:
Updated.